xskillscore/core/comparative.py

```diff
@@ -167,3 +177,141 @@ def _categorical_metric(observations, forecasts, dim):
+            * if ``alpha`` is float: half-width of the confidence interval at the
+             significance level ``alpha``.
+            * if ``alpha == 'return_p'``: the significance level that forecasts1 is
+             different than forecasts2.
+
+    Examples
+    --------
+    >>> f1 = xr.DataArray(np.random.normal(size=(30)),
+    ...      coords=[('time', np.arange(30))])
+    >>> f2 = xr.DataArray(np.random.normal(size=(30)),
+    ...      coords=[('time', np.arange(30))])
+    >>> o = xr.DataArray(np.random.normal(size=(30)),
+    ...      coords=[('time', np.arange(30))])
+    >>> mae_test(f1, f2, o, time_dim='time', dim=[], alpha=0.05)
+    <xarray.DataArray (results: 3)>
+    array([-0.09949737,  0.24421034,  0.        ])
+    Coordinates:
+        alpha    float64 0.05
+      * results  (results) <U11 'diff' 'hwci' 'significant'
+    >>> # absolute magnitude of difference is smaller than half-width of
+    >>> # confidence interval, therefore not significant at level alpha=0.05
+    >>> #
+    >>> mae_test(f1, f2, o, time_dim='time', dim=[], alpha='return_p')
+    <xarray.DataArray (results: 2)>
+    array([-0.09949737,  0.9226177])
+    Coordinates:
+      * results  (results) <U5 'diff' 'significance'
+    >>> # significance level of 0.92 much above commonly accepted level of 0.05
+
+
+    References
+    ----------
+        * https://www.cawcr.gov.au/projects/verification/CIdiff/FAQ-CIdiff.html
+
+    """
+    if isinstance(dim, str):
+        dim = [dim]
+    if time_dim in dim:
+        raise ValueError("`dim` cannot contain `time_dim`")
+
+    msg = f"`alpha` must be between 0 and 1 or `return_p`, found {alpha}"
+    if isinstance(alpha, str):
+        if alpha != "return_p":
+            raise ValueError(msg)
+    elif isinstance(alpha, float):
+        if alpha >= 1 or alpha <= 0:
+            raise ValueError(msg)
+    else:
+        raise ValueError(msg)
+
+    if observations is not None:
+        # Compare the forecasts and observations using metric
+        mae_f1o = mae(observations, forecasts1, dim=dim)
+        mae_f2o = mae(observations, forecasts2, dim=dim)
+    elif observations is None:
+        mae_f1o = forecasts1
+        mae_f2o = forecasts2
+
+    pearson_r_f1f2 = pearson_r(mae_f1o, mae_f2o, dim=time_dim)
+
+    # diff mae
+    diff = (mae_f2o - mae_f1o).mean(time_dim)
+
+    notnan = 1 * (mae_f1o.notnull() & mae_f2o.notnull())
+    N = notnan.sum(time_dim)
+    # average variances and take square root instead of averaging standard deviations
+    std = ((mae_f1o.var(time_dim) + mae_f2o.var(time_dim)) / 2) ** 0.5
+
+    if not isinstance(alpha, str):
+        confidence = st.norm.ppf(1 - alpha / 2)
+        # half width of the confidence interval
+        hwci = (2 * (1 - pearson_r_f1f2)) ** 0.5 * confidence * std / N ** 0.5
+        diff_greater_hwci = np.abs(diff) > hwci  # difference between MAEs significant?
+        ret = xr.concat([diff, hwci, diff_greater_hwci], "results")
```

Comment:
@ahuang11 @bradyrx @raybellwaves @dougiesquire do you have an opinion on the output format? To return objects or one concatened? Significant is actually not needed.

Pros concatinated: can combine more than two returns easily
Pros two returns: no need to come up with new dimension name, now generic ‚results‘

General: all tests in comparative should have aligned API and returns, similar to how metrics inputs and one output are all aligned

Reply:
concatenated probably.

Reply:
I though it would be nice to get a p value returned...

Reply:
I think if you look at any `scipy` or `numpy` statistical function, they always return separate objects. Granted, they are unlabeled arrays and don't have the opportunity to do labeled datasets. But I tend to think it's probably better to return a single dataarray (or dataset if multi-variate) for each quantified assessment. 

E.g., `linregress` offers 5 output arrays: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.linregress.html. `pearsonr` has `r, p = pearsonr(...)`: https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.pearsonr.html.

Reply:
Ok. But are you fine with the second return depending on the alpha?

Reply:
Or I try to add p as coord